### PR TITLE
SRV-122 Removing required payments from Advertiser

### PR DESCRIPTION
**Pull request by teghnet**

### CODE REVIEW

File: app/Http/Controllers/SupplyController.php [hunk 1]
```diff
@@ -100,9 +100,9 @@ public function find(
     public function findScript(Request $request): StreamedResponse
     {
         $params = [
-            json_encode(config('app.adserver_host')),
-            json_encode(config('app.aduser_external_location')),
-            json_encode('.'.config('app.website_banner_class')),
+            config('app.adserver_banner_host'),
+            config('app.aduser_external_location'),
+            '.'.config('app.website_banner_class'),
```
Comment (c3zi) [inline app/Http/Controllers/SupplyController.php:105]: Hmmm.. I think that `adserver_host` and `adserver_banner_host` can be different addresses.
Adserver may store banners on EC2 or somewhere. In this situation, Adserver has a different address for his host and banners. This one (host) can be used for `register` path but adserver_banner_host can be used only during the creation of banner URL.

File: app/Console/Commands/AdPayGetPayments.php [hunk 2]
```diff
@@ -35,20 +38,39 @@ public function handle(AdPay $adPay): void
     {
         $calculations = collect($adPay->getPayments(now()->getTimestamp()));
 
-        $this->info('Found '.count($calculations).' calculations.');
+        Log::info('Found '.count($calculations).' calculations.');
 
         $idList = $calculations->map(function (array $amount) {
```
Comment (c3zi) [inline app/Console/Commands/AdPayGetPayments.php]: `$idList` --> `$eventIds`

File: app/Console/Commands/AdPayGetPayments.php [hunk 3]
```diff
@@ -35,20 +38,39 @@ public function handle(AdPay $adPay): void
...
             return hex2bin($amount['event_id']);
         });
 
         $entries = EventLog::whereIn('event_id', $idList)
```
Comment (c3zi) [inline app/Console/Commands/AdPayGetPayments.php]: `$entries` --> `$unpaidEvents`

File: app/Console/Commands/AdPayGetPayments.php [hunk 4]
```diff
@@ -35,20 +38,39 @@ public function handle(AdPay $adPay): void
...
-            ->get()
-            ->each(function (EventLog $entry) use ($calculations) {
-                $calculation = $calculations->firstWhere('event_id', $entry->event_id);
+            ->whereNull('event_value')
+            ->get();
+
+        Log::info('Found '.count($entries).' entries that require values.');
 
-                $entry->update(['event_value' => $calculation['amount']]);
+        $ledgerEntries = $entries->groupBy(function (EventLog $entry) {
```
Comment (c3zi) [inline app/Console/Commands/AdPayGetPayments.php]: `$ledgerEntries` --> `$ledgerUnpaidEvents`

File: app/Console/Commands/AdPayGetPayments.php [hunk 5]
```diff
@@ -35,20 +38,39 @@ public function handle(AdPay $adPay): void
...
+            $totalEventValue = $collection->sum('event_value');
+
+            $userLedgerEntry = new UserLedgerEntry();
+            $userLedgerEntry->user_id = $userId;
+            $userLedgerEntry->amount = -$totalEventValue;
+            $userLedgerEntry->status = UserLedgerEntry::STATUS_ACCEPTED;
+            $userLedgerEntry->type = UserLedgerEntry::TYPE_AD_EXPENDITURE;
+            $userLedgerEntry->save();
+
+            return $userLedgerEntry;
```
Comment (c3zi) [inline app/Console/Commands/AdPayGetPayments.php:74]: We should use `transaction` to ensure consistency ;/

**Review by c3zi — CHANGES_REQUESTED** [review]

File: app/Console/Commands/AdPayGetPayments.php (hunk 5 above)
Comment (teghnet) [reply app/Console/Commands/AdPayGetPayments.php:74]: done

File: app/Console/Commands/AdPayGetPayments.php (hunk 4 above)
Comment (teghnet) [reply app/Console/Commands/AdPayGetPayments.php]: done

File: app/Http/Controllers/SupplyController.php (hunk 1 above)
Comment (teghnet) [reply app/Http/Controllers/SupplyController.php:105]: done

File: app/Console/Commands/AdPayGetPayments.php (hunk 5 above)
Comment (teghnet) [reply app/Console/Commands/AdPayGetPayments.php:74]: done

**Review by c3zi — APPROVED** [review]

**Pull request merged**